File: nom-api/Nom.Orch/Models/Shopping/RetailPackagingModel.cs
namespace Nom.Orch.Models.Shopping
{
    public class RetailPackagingResponseModel
    {
        public long Id { get; set; }
        public string IngredientPattern { get; set; } = string.Empty;
        public string PackageName { get; set; } = string.Empty;
        public decimal PackageSize { get; set; }
        public string PackageSizeUnit { get; set; } = string.Empty;
        public string SizeCategory { get; set; } = string.Empty;
        public decimal SizeInBaseUnits { get; set; }
        public bool IsDefault { get; set; }
        public string Source { get; set; } = string.Empty;
    }

    public class RetailPackagingCreateModel
    {
        public string IngredientPattern { get; set; } = string.Empty;
        public string PackageName { get; set; } = string.Empty;
        public decimal PackageSize { get; set; }
        public string PackageSizeUnit { get; set; } = string.Empty;
        public string SizeCategory { get; set; } = string.Empty;
        public decimal SizeInBaseUnits { get; set; }
        public bool IsDefault { get; set; } = true;
        public string Source { get; set; } = "manual";
    }

    public class RetailPackagingUpdateModel
    {
        public string? PackageName { get; set; }
        public decimal? PackageSize { get; set; }
        public string? PackageSizeUnit { get; set; }
        public string? SizeCategory { get; set; }
        public decimal? SizeInBaseUnits { get; set; }
        public bool? IsDefault { get; set; }
    }

    public class RetailPackagingLookupRequest
    {
        public List<string> IngredientNames { get; set; } = new();
    }

    public class RetailPackagingLookupResponse
    {
        public List<RetailPackagingResponseModel> Results { get; set; } = new();
        public List<string> NotFound { get; set; } = new();
        public bool AiLookupPerformed { get; set; }
    }

    /// <summary>
    /// Internal model for parsing AI JSON output.
    /// </summary>
    public class AiRetailPackagingSuggestion
    {
        public string IngredientPattern { get; set; } = string.Empty;
        public string PackageName { get; set; } = string.Empty;
        public decimal PackageSize { get; set; }
        public string PackageSizeUnit { get; set; } = string.Empty;
        public string SizeCategory { get; set; } = string.Empty;
    }
}

File: nom-ui-new/src/app/core/models/retail-packaging.model.ts
export interface RetailPackagingResponse {
  id: number;
  ingredientPattern: string;
  packageName: string;
  packageSize: number;
  packageSizeUnit: string;
  sizeCategory: 'volume' | 'mass' | 'count';
  sizeInBaseUnits: number;
  isDefault: boolean;
  source: string;
}

export interface RetailPackagingLookupRequest {
  ingredientNames: string[];
}

export interface RetailPackagingLookupResponse {
  results: RetailPackagingResponse[];
  notFound: string[];
  aiLookupPerformed: boolean;
}

export interface RetailPackagingCreateRequest {
  ingredientPattern: string;
  packageName: string;
  packageSize: number;
  packageSizeUnit: string;
  sizeCategory: string;
  sizeInBaseUnits?: number;
  isDefault?: boolean;
  source?: string;
}
